### PR TITLE
#182282684 - SET UP REACT PROJECT

### PR DESCRIPTION
**Pull request by dalienst**
#### What does this Pull Request do?

- Set up of react project
- set up of ESlint
- set up of Prettier formatter
- set up of PR template

#### Description of Task to be completed?

- Setting up frontend environment of the application

#### How should this be manually tested?

- run
```
npm start
```

#### Any background context you want to provide?

- Code having a standard format before being merged into main

#### What are the relevant pivotal tracker?

- [182282684](https://www.pivotaltracker.com/n/projects/2570600/stories/182282684)

#### Screenshots (if appropriate)

-
image

#### Questions:

- N/A

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "env": {
+        "browser": true,
+        "es2021": true
```
Comment (ChrisOnsando) [inline .eslintrc.json:4]: include "jest": true, in the env

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Comment (ChrisOnsando) [inline public/index.html:27]: You could change this to Speaskfer

File: .eslintrc.json (hunk 1 above)
Comment (ChrisOnsando) [inline .eslintrc.json:4]: Include "jest":true, if we are testing

Comment (ChrisOnsando) [inline .eslintrc.json]: Include "jest": true, in the env if we will be testing

File: public/index.html (hunk 2 above)
Comment (ChrisOnsando) [inline public/index.html]: You could change this to Speaksfer

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    ],
+    "overrides": [
+    ],
+    "parserOptions": {
+        "ecmaVersion": "latest",
+        "sourceType": "module"
+    },
+    "plugins": [
+        "react"
+    ],
```
Comment (KaigaMunene) [inline .eslintrc.json:23]: add prettier and jest

File: .eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{
+    "env": {
+        "browser": true,
+        "es2021": true
+    },
+    "extends": [
+        "plugin:react/recommended",
+        "airbnb",
+        "plugin:prettier/recommended"
+    ],
```
Comment (KaigaMunene) [inline .eslintrc.json:12]: add a plugin for jest in extends i.e     "plugin:jest/recommended"

**Review by KaigaMunene — COMMENTED** [review]
- In the PR message, on How should this be manually tested?
write the command to do this
- Also in the PR message put an appropriate screenshot to show that the setup is working
- Add an ignore file for both eslint and prettier, that is .eslintignore and .prettierignore

File: README.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
+# Getting Started with Create React App
```
Comment (narravabrion) [inline README.md]: You can just delete everything in the README file.

**Comment by MwanikiN** [comment]
Reviewed

File: .eslintrc.json (hunk 3 above)
Comment (dalienst) [reply .eslintrc.json:23]: Done

File: public/index.html (hunk 2 above)
Comment (dalienst) [reply public/index.html]: Done

File: .eslintignore [hunk 6]
```diff
@@ -0,0 +1,3 @@
+build
+node_modules
+.github
```
Comment (KaigaMunene) [inline .eslintignore]: remove .github

File: .eslintrc.json [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        "sourceType": "module"
+    },
+    "plugins": [
+        "react",
+        "prettier",
+        "jest"
+    ],
+    "rules": {
+        "react/react-in-jsx-scope": "off",
+        "prettier/prettier": 2
```
Comment (KaigaMunene) [inline .eslintrc.json]: why ?
i think it should be "prettier/prettier": "error"

File: .eslintignore (hunk 6 above)
Comment (dalienst) [reply .eslintignore]: Done

File: .prettierignore [hunk 8]
```diff
@@ -0,0 +1,3 @@
+build
+node_modules
+.github
```
Comment (KaigaMunene) [inline .prettierignore]: remove .github

File: .eslintignore [hunk 9]
```diff
@@ -0,0 +1,2 @@
+build
+node_modules
```
Comment (mainamuriuki274) [inline .eslintignore]: Add empty line after this file

File: .gitignore [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+# testing
+/coverage
+
+# production
+/build
+
+# misc
+.DS_Store
+.env.local
```
Comment (mainamuriuki274) [inline .gitignore:16]: add .env

File: .husky/pre-commit [hunk 11]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npx lint-staged
```
Comment (mainamuriuki274) [inline .husky/pre-commit]: add empty line after this

File: .lintstagedrc.json [hunk 12]
```diff
@@ -0,0 +1,10 @@
+{
+    "*.js":[
+        "prettier --write",
+        "eslint"
+    ],
+    "*.jsx":[
+        "prettier --write",
+        "eslint"
+    ]
+}
```
Comment (mainamuriuki274) [inline .lintstagedrc.json]: add empty line after this file

File: .prettierignore (hunk 8 above)
Comment (mainamuriuki274) [reply .prettierignore]: It can remain so that prettier ignores it

Comment (mainamuriuki274) [inline .prettierignore]: add empty line after this

File: .prettierrc [hunk 13]
```diff
@@ -0,0 +1,10 @@
+{
+  "semi": true,
+  "tabWidth": 2,
+  "printWidth": 100,
+  "singleQuote": true,
+  "trailingComma": "none",
+  "jsxBracketSameLine": true,
+  "useTabs": false,
+  "endOfLine": "lf"
+}
```
Comment (mainamuriuki274) [inline .prettierrc]: add empty line this

File: package.json [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+    "eslint-config-google": "^0.14.0",
+    "eslint-config-prettier": "^8.5.0",
+    "eslint-plugin-import": "^2.26.0",
+    "eslint-plugin-jsx-a11y": "^6.6.1",
+    "eslint-plugin-prettier": "^4.2.1",
+    "eslint-plugin-react": "^7.30.1",
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "husky": "^8.0.1",
+    "prettier": "^2.7.1"
+  }
```
Comment (mainamuriuki274) [inline package.json:63]: add node and npm versions

**Review by mainamuriuki274 — CHANGES_REQUESTED** [review]

File: .eslintignore (hunk 9 above)
Comment (dalienst) [reply .eslintignore]: Okay

**Review by mainamuriuki274 — APPROVED** [review]

**Review by 3N61N33R — APPROVED** [review]

**Pull request merged**